kong/router.lua

```diff
@@ -622,6 +623,11 @@ local function marshall_route(r)
         if type(sni) ~= "string" then
           return nil, "sni elements must be strings"
         end
+        
+        if sni:len() > 1 and sni:sub(-1) == "." then
+          log(DEBUG, "last dot in FQDNs must not be used for routing, removing in ", sni)
+          sni = sni:sub(1, -2)
```

Comment:
Does the `trailing slash` have meaning elsewhere outside `routing`? Just thinking that can we hit this issue elsewhere? Or is it just about routing?

Reply:
Ah, trailing dot makes it FQDN, otherwise it may be appended with, well whatever in search.